src/parser.js

```diff
@@ -105,7 +105,8 @@ const onMetadata = (metadata, infohash, knex) => {
 		const names = files.map(({ path }) => (Array.isArray(path) ? path.join('/') : path)).concat(name);
 		const invalid = filterTorrent(names);
 		const filesWithOriginal = name && length ? [{ length, path: name }, ...files] : files;
-		if (!invalid.length>0 && length>0){
+
+		if (!invalid.length > 0 && files.length > 0) {
```

Comment:
if there are no files, this will fail

Try `if (!invalid.length > 0 && filesWithOriginal.length > 0) {`

Reply:
Good catch thanks